node/src/main/scala/org/bitcoins/node/Node.scala

```diff
@@ -147,6 +170,90 @@ trait Node extends NodeApi with ChainQueryApi with P2PLogger {
   def isDisconnected(idx: Int): Future[Boolean] =
     peerMsgSenders(idx).isDisconnected()
 
+  private def createInDbIfBlockFilterPeer(peer: Peer): Future[Unit] = {
+    if (peerData(peer).serviceIdentifier.nodeCompactFilters) {
+      logger.info(s"Our peer=$peer has been initialized")
+      PeerDAO().upsertPeer(
+        s"${peer.socket.getHostString}:${peer.socket.getPort}")
+    } else {
+      logger.info(
+        s"Our peer=$peer does not support compact filters. Disconnecting.")
+      PeerDAO().deleteByKey(s"${peer.socket.getHostString}:${peer.socket.getPort}")
+      peerData(peer).peerMessageSender.disconnect()
+    }
+    Future.unit
+  }
+
+  def createInDbIfBlockFilterPeer(networkAddress: NetworkIpAddress): Unit = {
+    val ipv4bytes: Option[ByteVector] =
+      try {
+        Some(networkAddress.address.ipv4Bytes)
+      } catch {
+        case _: Throwable =>
+          logger.info("Ignoring ipv6 address from addr message")
+          return
+      }
+    val stringAddress = ipv4bytes.get.toArray
+      .map(x => {
+        val short = x.toShort
+        if (short < 0) x + 256
+        else short
+      })
+      .mkString(".") + s":${networkAddress.port}"
+    if (networkAddress.services.nodeCompactFilters) {
+      logger.info(s"Peer from addr: $stringAddress supports compact filters.")
+      PeerDAO().upsertPeer(stringAddress)
+    }
+    ()
+  }
+
+  def createInDbIfBlockFilterPeer(addr: IPv4AddrV2Message): Unit = {
```

Comment:
the same comment above applies to this, this is `Neutrino` specific, so we should move to `NeutrinoNode`